Objects/templateobject.c

```diff
@@ -114,7 +114,10 @@ template_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
             last_was_str = 0;
         }
         else {
-            PyErr_SetString(PyExc_TypeError, "Template.__new__ *args need to be of type 'str' or 'Interpolation'");
+            PyErr_Format(
+                PyExc_TypeError,
+                "Template.__new__ *args need to be of type 'str' or 'Interpolation', got %.200s",
+                Py_TYPE(item)->tp_name);
```

Comment:
Maybe this is a bit better?

```suggestion
                "Template.__new__ *args need to be of type 'str' or 'Interpolation', got %T",
                item);
```